code/modules/mob/living/simple_animal/abnormality/he/singing_machine.dm

```diff
@@ -130,49 +161,96 @@ Finally, an abnormality that DOESN'T have to do any fancy movement shit. It's a
 	return ..()
 
 /mob/living/simple_animal/hostile/abnormality/singing_machine/ZeroQliphoth(mob/living/carbon/human/user) // WARNING: Don't call this on its own. Several zero-qliphoth behaviors rely on its qliphoth actually being 0.
-	icon_state = "singingmachine_open_[cleanliness]" // Machine opens and starts making horrible empty grinding noises.
-	icon_living = icon_state
-	update_icon()
 	playsound(src, 'sound/abnormalities/singingmachine/open.ogg', 200, 0, playRange)
+	machine_hinge.moveHinge(-90, 1.5 SECONDS, -2, 6)
+	particle_note_broken = new(get_turf(src))
+	particle_grind = new(get_turf(src))
 	grindNoise = new(list(src), TRUE)
-	playStatus = 1
+	playStatus = GRINDING
 	return
 
+/mob/living/simple_animal/hostile/abnormality/singing_machine/proc/noiseEffect()
+	for(var/mob/living/carbon/human/H in livinginrange(playRange, src))
+		if(faction_check_mob(H))
+			continue
+		if(playStatus == GRINDING) // If it's mad, damages everyone in it's radius.
+			H.apply_damage(rand(playStatus * noiseFactor, playStatus * noiseFactor * 2), WHITE_DAMAGE, null, H.run_armor_check(null, WHITE_DAMAGE), spread_damage = TRUE)
+			to_chat(H, span_warning("That terrible grinding noise..."))
+			continue
+		if(H in musicalAddicts) // If it's playing, it spreads the status effect to everyone.
+			continue
+		addAddict(H)
+
+/mob/living/simple_animal/hostile/abnormality/singing_machine/proc/damageAddicts()
+	for(var/mob/living/carbon/human/addict in musicalAddicts)
+		addict.apply_damage(rand(playStatus * noiseFactor, playStatus * noiseFactor * 2), WHITE_DAMAGE, null, addict.run_armor_check(null, WHITE_DAMAGE), spread_damage = TRUE)
+		to_chat(addict, span_warning("You can hear it again... it needs more..."))
+		if(!addict.sanity_lost)
+			continue
+		applySpecialInsanity(addict)
+
+/mob/living/simple_animal/hostile/abnormality/singing_machine/proc/addAddict(mob/living/carbon/human/target)
+	target.apply_status_effect(STATUS_EFFECT_MUSIC)
+	to_chat(target, span_nicegreen("There's something about that sound..."))
+	musicalAddicts |= target
+	SEND_SOUND(target, 'sound/abnormalities/singingmachine/addiction.ogg')
+	addtimer(CALLBACK(src, .proc/removeAddict, target), 5 MINUTES)
+
 /mob/living/simple_animal/hostile/abnormality/singing_machine/proc/removeAddict(mob/living/carbon/human/addict)
 	if(addict)
 		musicalAddicts -= addict // Your five minutes are over, you're free.
 
+/mob/living/simple_animal/hostile/abnormality/singing_machine/proc/startPlaying()
```

Comment:
```suggestion
/mob/living/simple_animal/hostile/abnormality/singing_machine/proc/startPlaying()
	if(playStatus == PLAYING)
		return FALSE
```
This prevents duplicate calls from making unending particle effects or interrupting a future instance.